kubernetes/structures_pod.go

```diff
@@ -1025,6 +1049,22 @@ func expandVolumes(volumes []interface{}) ([]v1.Volume, error) {
 	return vl, nil
 }
 
+func expandReadinessGates(gates []interface{}) ([]v1.PodReadinessGate, error) {
+	if len(gates) == 0 {
```

Comment:
Here you also need to check for `gates[0] = nil` (see the other TypeList expanders).
I know it's not intuitive but if you set it to empty block in configuration, you get a nil value and it will crash the provider.
```
readiness_gate {}
```

Reply:
Done